BUGFIX: Always bundle custom svgs instead of using `resource://` paths

mhsdesign opened the pull request:
Resolves: https://github.com/neos/neos-ui/issues/3616
Resolves: https://github.com/neos/neos-ui/issues/3712

See https://github.com/neos/neos-ui/issues/2092#issuecomment-2283775887 for an explanation

**What I did**

The custom icons are now included into the bundle instead of requested by path.

<img width="445" alt="image" src="https://github.com/user-attachments/assets/d7d9e397-586a-4229-aff2-d311abb708f8">

Also to unify things and simplify the build-stack, the Neos icon is now also bundled as plain svg instead of data-url:

<img width="219" alt="image" src="https://github.com/user-attachments/assets/09cc100c-51d2-4bb8-966d-2ca292b2b272">


Also the fallback image of the image editor will be fixed for cloud hosted resources now:

| before | after |
|--------|--------|
| <img width="317" alt="image" src="https://github.com/user-attachments/assets/96f3aadc-a3cc-4d99-859a-78ea8d8d5a2d"> | <img width="322" alt="image" src="https://github.com/user-attachments/assets/c60b2cc3-108f-4434-93ba-8c376011a459"> | 

Surprisingly i could not observer any pixel shifting. So this should be 100% the same visual look.



**How I did it**

**How to verify it**

<!--
If possible, a screenshot or a gif comparing the new and old behavior would be great.
-->

mhsdesign committed:
BUGFIX: Always bundle custom svgs instead of using `resource://` paths

Sebobo commented on packages/react-ui-components/src/Icon/resourceIcon.tsx:
```diff
@@ -13,6 +13,11 @@ export interface ResourceIconProps extends Omit<IconProps, 'theme'> {
 }
 
+/**
+ * @deprecated please refrain from using resource paths
```
The `Icon` component is still using the ResourceIcon when defining custom icons in yaml. What is the intended replacement?

Sebobo reviewed (changes requested):
Works and looks great, but I don't see yet how we can deprecate the now old `ResourceIcon` if the new one does something else and from what I see doesn't support custom icons.

mhsdesign committed:
BUGFIX: Include the image editors dummy image in the bundle

using `/_Resources/Static/Packages/Neos.Neos/Images/dummy-image.svg` is a bad bet as for cloud hosted resources this wont work and it will show a dead-fallback image.

mhsdesign commented:
As written in https://github.com/neos/neos-ui/issues/2092

> Case (3.) is more complicated, I'm afraid. The `<ResourceIcon/>` is a specialization of the (unfortunately public API) `<Icon/>`-component. We will need to deprecate its behavior, and (prior to that) find a way to allow for custom Icon URIs that can be resolved on the server side (we'll some sort of configurable mapping thing here).

My idea is to deprecate the behaviour but thats just for us and without real effect obviously.
To get rid of all usages we will - and i will plan do so - make the `NodeTypeSchemaBuilder` transform all `resource://` paths on the server. 

That will leave us with no more usages of client `resource://` paths. Wdyt?

Sebobo reviewed (approved):
Works on my machine <3

mhsdesign commented on packages/neos-ui-editors/src/Editors/Image/resource/dummy-image.dataurl.svg:
This is now duplicated here and in `Neos.Neos` but yeah there is no way around this ... and it somehow belongs here imo

mhsdesign committed:
TASK: Optimize Neos logo.svg

see discussion in Neos.Neos where we did the same https://github.com/neos/neos-development-collection/pull/5126/files#r1625874465

- remove obsolete `g` tags (and german illustrator references :D)
- inline style

grebaldi reviewed (approved):
Looks & works very well :)
